### PR TITLE
Switch from basic authentication to jwt authentication

### LINKED ISSUE #15: Implement Oauth with JWTs
Currently we are using Basic Authentication which is insecure over http.

### PR DESCRIPTION
**Pull request by Green-Ranger11**
This PR switches authentication from basic to jwt.
This closes #15

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (nfebe) [inline .eslintrc.json]: There's an issue for this, you want to take this linting update to its own PR.

File: Controllers/AuthenticationController.js [hunk 2]
```diff
@@ -1,31 +1,58 @@
...
+    res.redirect('/');
+  },
+  createUser: (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      res.status(400).json({ errors: errors.array() });
+    }
+    const { email } = req.body.user;
+    // eslint-disable-next-line consistent-return
+    User.findOne({ email }).then((foundUser) => {
```
Comment (nfebe) [inline Controllers/AuthenticationController.js]: `foundUser` => `user`

File: Controllers/PublicController.js [hunk 3]
```diff
@@ -1,7 +1,5 @@
 export default {
-    welcome : (req, res) => {
-    res.json({ 'welcome': 'Welcome to the StandupMan API!!' });
-    }
+  welcome: (req, res) => {
+    res.json({ welcome: 'Welcome to the StandupMan API!!' });
+  },
```
Comment (nfebe) [inline Controllers/PublicController.js]: Do linting fixes in the issue that adds the linter updates.

File: Controllers/StandUpController.js [hunk 4]
```diff
@@ -1,149 +1,166 @@
-import StandUp from '../Models/StandUp.js'
-import StandUpUpdate from '../Models/StandUpUpdate.js'
-import User from '../Models/User.js'
+/* eslint-disable class-methods-use-this */
 import { validationResult } from 'express-validator';
+import StandUp from '../Models/StandUp';
```
Comment (nfebe) [inline Controllers/StandUpController.js]: Same here for this linting changes

**Review by nfebe — CHANGES_REQUESTED** [review]
Please take out all linting changes and linter introduction to seperate pr, and leave only jwt updates here!

Thank you! :)

File: .env.example [hunk 5]
```diff
@@ -1,3 +1,4 @@
 APP_PORT=5000
 DB_HOST=127.0.0.1
-DB_NAME=scrum_app
\ No newline at end of file
+DB_NAME=scrum_app
+API_KEY=super_secret_key
```
Comment (nfebe) [inline .env.example]: JWT_SECRET_KEY

**Review by nfebe — CHANGES_REQUESTED** [review]
Looks good! Please, adjust minor change.

**Comment by Green-Ranger11** [comment]
Hi @Fenn-CS sorry to ping in case you are busy but just wanted to ask for your review again.

**Review by nfebe — APPROVED** [review]

**Comment by nfebe** [comment]
Hello @Green-Ranger11, sorry for the delays! Merged. Thank you :)

**Pull request merged**